[2.x] `scripted lm-coursier/*` are flaky

On Java 8:

```
Error:  x lm-coursier/update-sbt-classifiers-2 [PENDING]
[info] Running lm-coursier/version-reconciliation
Error:  Exception in thread "sbt-parser-init-thread" java.lang.ExceptionInInitializerError
Error:  	at sbt.internal.parser.SbtParserInit$$anon$1.run(SbtParser.scala:179)
Error:  Caused by: java.util.ConcurrentModificationException
Error:  	at java.util.Hashtable$Enumerator.next(Hashtable.java:1408)
Error:  	at scala.collection.convert.JavaCollectionWrappers$JPropertiesWrapper$$anon$6.next(JavaCollectionWrappers.scala:584)
Error:  	at scala.collection.convert.JavaCollectionWrappers$JPropertiesWrapper$$anon$6.next(JavaCollectionWrappers.scala:580)
Error:  	at scala.collection.IterableOnceOps.find(IterableOnce.scala:633)
Error:  	at scala.collection.IterableOnceOps.find$(IterableOnce.scala:630)
Error:  	at scala.collection.AbstractIterator.find(Iterator.scala:1300)
Error:  	at dotty.tools.dotc.config.PathResolver$Environment$.searchForBootClasspath(PathResolver.scala:40)
Error:  	at dotty.tools.dotc.config.PathResolver$Environment$.javaBootClassPath(PathResolver.scala:49)
Error:  	at dotty.tools.dotc.config.PathResolver$Defaults$.javaBootClassPath(PathResolver.scala:75)
Error:  	at dotty.tools.dotc.config.CommonScalaSettings.$init$(ScalaSettings.scala:96)
Error:  	at dotty.tools.dotc.config.ScalaSettings.<init>(ScalaSettings.scala:15)
Error:  	at dotty.tools.dotc.core.Contexts$ContextBase.<init>(Contexts.scala:847)
Error:  	at dotty.tools.dotc.Driver.initCtx(Driver.scala:61)
Error:  	at sbt.internal.parser.SbtParser$ParseDriver.<init>(SbtParser.scala:138)
Error:  	at sbt.internal.parser.SbtParser$.<clinit>(SbtParser.scala:135)
Error:  	... 1 more
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -49,11 +49,11 @@ jobs:
             distribution: zulu
             jobtype: 9
           - os: ubuntu-latest
-            java: 8
+            java: 11
             distribution: temurin
             jobtype: 10
           - os: windows-latest
-            java: 8
+            java: 11
             distribution: temurin
             jobtype: 10
     runs-on: ${{ matrix.os }}

